git/archive.go

```diff
@@ -0,0 +1,274 @@
+package git
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/flate"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+)
+
+type ArchiveOptions struct {
+	Verbose            bool
+	List               bool
+	WorktreeAttributes bool
+	ArchiveFormat      string
+	BasePrefix         string
+	OutputFile         string
+	CompressionLevel   int
+}
+
+type ArchiveFormat int
+
+const (
+	archiveUnknow = ArchiveFormat(iota)
```

Comment:
archiveUnknown

Reply:
Yeah.. This is because I'm very used to code in C++ (only one return statement). 
I'll update the code to be more golang friendly :+1: